File: SpaceCommandR/Server/Classes/Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SpaceCommandR.Server.Classes
{
    public class Player
    {
        public int id { get; set; }
        public string name { get; set; }
        public string color { get; set; }
        public bool isAlive { get; set; }
        public bool isActive { get; set; }
        public int score { get; set; }
        public int energy { get; set; }
        public int lastSelectedStation { get; set; }

        public void Update(Player player)
        {
            this.color = player.color;
            this.isAlive = player.isAlive;
            this.isActive = player.isActive;
            this.score = player.score;
            this.energy = player.energy;
            this.lastSelectedStation = player.lastSelectedStation;
        }
    }
}

File: SpaceCommandR/Client/Player/player.js
var Game = Game || {};
Game.Player = (function(Game){

	/*
	*
	* constructor
	*
	*/

	function Player (options){
	    this.id = options.id;
	    this.name = options.name;
	    this.color = options.color || "green";
	    this.isAlive = options.isAlive || false;
	    this.isActive = options.isActive || false;
	    this.score = 0;
	    this.energy = 0;

	    this.lastSelectedStation = 0;
	}

	Player.prototype.destroy = function () {
	    this.isAlive = false;
	    toastr.info(this.name + " destroyed.");
	};

	Player.prototype.getPosition = function (max_x, max_y) {
	    var coords = {};
	    switch (this.id) {
	        case 1:
	            coords = { x: 100, y: 100 };
	            break;
	        case 2:
	            coords = { x: max_x - 100, y: max_y - 100 };
	            break;
	        case 3:
	            coords = { x: max_x - 100, y: 100 };
	            break;
	        case 4:
	            coords = { x: 100, y: max_y - 100 };
	            break;
	        default:
	            coords = { x: 100, y: 100 };
	            break;
	    }
	    return coords;
	};

	return Player;
}(Game));